### PR TITLE
fs: rename block to cluster

### PR DESCRIPTION
**Pull request by wmitros**

### CODE REVIEW

**Comment by varqox** [comment]
Note for merger: This PR should be **merged after** #47.

**Review by psarna — COMMENTED** [review]
Please add a short changelog and motivation in the commit message. Renaming from block to cluster is fine, but ` fs: Refactored cluster_allocator ` is not enough - let's write a line or two that optional is chosen instead of 0 and there's a deque now.

File: tests/unit/cluster_allocator_test.cc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+BOOST_AUTO_TEST_CASE(cluster_allocator_empty) {
+    std::unordered_set<fs::cluster_id_t> empty_uset;
+    fs::cluster_allocator empty_ca{empty_uset, {}};
+    BOOST_REQUIRE(empty_ca.alloc() == std::nullopt);
+}
+
+BOOST_AUTO_TEST_CASE(cluster_allocator_small) {
+    std::unordered_set<fs::cluster_id_t> empty_uset;
+    std::deque<fs::cluster_id_t> deq{1, 5, 3, 4, 2};
+    fs::cluster_allocator small_ca(empty_uset, deq);
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:53]:
```suggestion
    fs::cluster_allocator small_ca({}, deq);
```

File: tests/unit/cluster_allocator_test.cc [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    small_ca.free(deq[2]);
+    small_ca.free(deq[1]);
+    small_ca.free(deq[3]);
+    small_ca.free(deq[0]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[4]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[2]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[1]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[3]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[0]);
+    BOOST_REQUIRE(small_ca.alloc() == std::nullopt);
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:67]:
```suggestion
    BOOST_REQUIRE(small_ca.alloc() == std::nullopt);

```

File: tests/unit/cluster_allocator_test.cc [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+BOOST_AUTO_TEST_CASE(cluster_allocator_small) {
+    std::unordered_set<fs::cluster_id_t> empty_uset;
+    std::deque<fs::cluster_id_t> deq{1, 5, 3, 4, 2};
+    fs::cluster_allocator small_ca(empty_uset, deq);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[0]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[1]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[2]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[3]);
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:57]:
```suggestion
    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[3]);

```

File: tests/unit/cluster_allocator_test.cc [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+BOOST_AUTO_TEST_CASE(cluster_allocator_max) {
+    constexpr fs::cluster_id_t clusters_per_shard = 1024;
+    std::deque<fs::cluster_id_t> dq;
+    std::unordered_set<fs::cluster_id_t> empty_uset;
+    for (fs::cluster_id_t i = 0; i < clusters_per_shard; i++) {
+        dq.emplace_back(i);
+    }
+    fs::cluster_allocator ordinary_ca(empty_uset, dq);
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:86]:
```suggestion
    fs::cluster_allocator ordinary_ca({}, dq);
```

File: tests/unit/cluster_allocator_test.cc [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    small_ca.free(deq[2]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[3]);
+    small_ca.free(deq[4]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[2]);
+}
+
+BOOST_AUTO_TEST_CASE(cluster_allocator_max) {
+    constexpr fs::cluster_id_t clusters_per_shard = 1024;
+    std::deque<fs::cluster_id_t> dq;
+    std::unordered_set<fs::cluster_id_t> empty_uset;
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:82]:
```suggestion
```

File: tests/unit/cluster_allocator_test.cc [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+BOOST_AUTO_TEST_CASE(cluster_allocator_empty) {
+    std::unordered_set<fs::cluster_id_t> empty_uset;
+    fs::cluster_allocator empty_ca{empty_uset, {}};
+    BOOST_REQUIRE(empty_ca.alloc() == std::nullopt);
+}
+
+BOOST_AUTO_TEST_CASE(cluster_allocator_small) {
+    std::unordered_set<fs::cluster_id_t> empty_uset;
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:51]:
```suggestion
```

File: tests/unit/cluster_allocator_test.cc [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[4]);
+    small_ca.free(deq[2]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[3]);
+    small_ca.free(deq[4]);
+    BOOST_REQUIRE_EQUAL(small_ca.alloc().value(), deq[2]);
+}
+
+BOOST_AUTO_TEST_CASE(cluster_allocator_max) {
+    constexpr fs::cluster_id_t clusters_per_shard = 1024;
+    std::deque<fs::cluster_id_t> dq;
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:81]:
```suggestion
    std::deque<fs::cluster_id_t> deq;
```

File: tests/unit/cluster_allocator_test.cc [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+    BOOST_REQUIRE(ca.alloc() == std::nullopt);
+    BOOST_REQUIRE(ca.alloc() == std::nullopt);
+    ca.free(0);
+    BOOST_REQUIRE_EQUAL(ca.alloc().value(), 0);
+    BOOST_REQUIRE(ca.alloc() == std::nullopt);
+    BOOST_REQUIRE(ca.alloc() == std::nullopt);
+}
+
+BOOST_AUTO_TEST_CASE(cluster_allocator_empty) {
+    std::unordered_set<fs::cluster_id_t> empty_uset;
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:45]:
```suggestion
```

File: tests/unit/cluster_allocator_test.cc [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+    fs::cluster_allocator empty_ca{empty_uset, {}};
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:46]:
```suggestion
    fs::cluster_allocator empty_ca{{}, {}};
```

File: tests/unit/cluster_allocator_test.cc [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+    constexpr fs::cluster_id_t clusters_per_shard = 1024;
+    std::deque<fs::cluster_id_t> dq;
+    std::unordered_set<fs::cluster_id_t> empty_uset;
+    for (fs::cluster_id_t i = 0; i < clusters_per_shard; i++) {
+        dq.emplace_back(i);
+    }
+    fs::cluster_allocator ordinary_ca(empty_uset, dq);
+    for (fs::cluster_id_t i = 0; i < clusters_per_shard; i++) {
+        BOOST_REQUIRE_EQUAL(ordinary_ca.alloc().value(), i);
+    }
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:89]:
```suggestion
    }
    BOOST_REQUIRE(ordinary_ca.alloc() == std::nullopt);
```

File: tests/unit/cluster_allocator_test.cc [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+    for (fs::cluster_id_t i = 0; i < clusters_per_shard; i++) {
+        ordinary_ca.free(i);
+    }
+}
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:93]:
```suggestion
}

```

File: tests/unit/cluster_allocator_test.cc [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+BOOST_AUTO_TEST_CASE(cluster_allocator_pseudo_rand) {
+    std::unordered_set<fs::cluster_id_t> uset;
+    std::deque<fs::cluster_id_t> dq;
```
Comment (varqox) [inline tests/unit/cluster_allocator_test.cc:96]:
```suggestion
    std::deque<fs::cluster_id_t> deq;
```

**Review by varqox — CHANGES_REQUESTED** [review]
Please correct these small code-style related issues

**Comment by psarna** [comment]
merging #47 created conflicts, please rebase

**Pull request merged**